### PR TITLE
Compiler warning fixes

### PR DESCRIPTION
**Pull request by jaydoane**
## Overview

Fix miscellaneous compiler warnings.

## Testing recommendations

```
make eunit
```

## Checklist

- [x] Code is written and works correctly
- [x] Changes are covered by tests
- [ ] Any new configurable parameters are documented in `rel/overlay/etc/default.ini`
- [ ] A PR for documentation changes has been made in https://github.com/apache/couchdb-documentation

### CODE REVIEW

File: src/mem3/src/mem3.erl [hunk 1]
```diff
@@ -323,8 +323,8 @@ range(#ordered_shard{range = Range}) ->
     Range;
 range(<<"shards/", Start:8/binary, "-", End:8/binary, "/", _/binary>>) ->
     [
-        httpd_util:hexlist_to_integer(binary_to_list(Start)),
-        httpd_util:hexlist_to_integer(binary_to_list(End))
+        list_to_integer(binary_to_list(Start), 16),
```
Comment (nickva) [inline src/mem3/src/mem3.erl:326]: We could even go a tiny bit faster by using `binary_to_integer/2`

```
./erlperf 'httpd_util:hexlist_to_integer(binary_to_list(<<"abc123def980">>)).' 'binary_to_integer(<<"abc123def980">>, 16).' --samples 10 --warmup 2
Code                                                                       ||        QPS       Time     Rel
binary_to_integer(<<"abc123def980">>, 16).                                  1   28289 Ki      35 ns    100%
httpd_util:hexlist_to_integer(binary_to_list(<<"abc123def980">>)).          1   20880 Ki      47 ns     73%
```

File: src/couch/src/couch_app.erl [hunk 2]
```diff
@@ -14,8 +14,6 @@
 
 -behaviour(application).
 
--include_lib("couch/include/couch_db.hrl").
-
```
Comment (nickva) [inline src/couch/src/couch_app.erl:18]: Nice! I've been seeing various "header not used" warnings from erlang_ls in emacs but short of opening every module I couldn't figure out a way to performing a scan for the whole project at once.

File: src/mem3/src/mem3.erl (hunk 1 above)
Comment (jaydoane) [reply src/mem3/src/mem3.erl:326]: Oh cool! Will update.

File: src/couch/src/couch_app.erl (hunk 2 above)
Comment (jaydoane) [reply src/couch/src/couch_app.erl:18]: Same. I really didn't want to open every file individually.

**Review by nickva — APPROVED** [review]
Very nice, +1

**Pull request merged**